junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/ExclusiveResource.java

```diff
@@ -29,6 +29,19 @@
 @API(status = EXPERIMENTAL, since = "1.3")
 public class ExclusiveResource {
 
+	/**
+	 * The key for the global resource lock that all direct children of the
+	 * engine descriptor acquire in {@linkplain LockMode#READ read} mode by
+	 * default.
+	 *
```

Comment:
Maybe extend the explanation a bit with the additional effects if any such child requests READ_WRITE for this key, i.e. lock coarsening to top-level descriptor, forced same thread exection

Reply:
Done in f7a7742ba8c4c8dab6956a2265262920e22b9434.